server/controllers/favorites.js

```diff
@@ -1,39 +1,50 @@
 import models from '../models';
 
+const { Favourite, Recipe, User } = models;
+
 export default {
   addUserFavorites(req, res) {
-    return models.Recipe
-      .findById(req.params.recipeId)
-      .then((recipe) => {
-        if (!recipe) {
-          return res.status(400).json({
-            message: 'Recipe does not exist'
+    return Favourite
+      .findOne({
+        where: {
+          userId: req.decoded.user.id,
+          recipeId: req.params.recipeId
+        }
+      }).then((favorite) => {
+        if (favorite) {
+          return res.status(404).json({
```

Comment:
Why do you decide to use `.json`?

Reply:
I used the `.json` method for specificity. Using `.json` also sets `content-type` in the response header to `application/json` The commonly used `.send` method is generic and can return different types of responses including json.